doc/mgmt/SONiC_YANG_Model_Guidelines.md

```diff
@@ -550,12 +562,166 @@ container sonic-interface {
 ......
 ```
 
+#### Example 2: Composite Keys with Same Number of Elements(NOT Allowed case)
+
+```yang
+......
+container sonic-interface {
+	container INTERFACE {
+		list INTERFACE_LIST {  // 1st list
+			key ifname;
+			leaf ifname {
+				type string;
+			}
+			// ...
+		}
+
+		list INTERFACE_ANOTHER_LIST { // Negative case
+			key ifname;
+			leaf ifname {
+				type string;
+			}
+			// ...
+		}
+	} 
+}
+......
+```
+
+#### Example 3: Pattern Constraints (Allowed case)
+
+In this example, `INTERFACE_LIST` uses a key `ifname` that must start with "Eth", while `INTERFACE_IPADDR_LIST` uses a key `ifname` that must start with "Vlan".
+
+```yang
+......
+container sonic-interface {
+	container INTERFACE {
+		list INTERFACE_LIST {  // 1st list
+			key ifname;
+			leaf ifname {
+				type string {
+					pattern "Eth.*";
+				}
+			}
+			// ...
+		}
+
+		list INTERFACE_IPADDR_LIST { // 2nd list
+			key "ifname ip_addr";
+			leaf ifname {
+				type string {
+					pattern "Vlan.*";
+				}
+			}
+			// ...
+		}
+	} 
+}
+......
+```
+
+#### Example 4: Pattern Constraints (NOT Allowed case)
+
+Here, both INTERFACE_LIST and INTERFACE_ANOTHER_LIST use a key ifname with the same pattern constraint, making them potentially overlapping and ambiguous.
+
+```yang
+......
+container sonic-interface {
+	container INTERFACE {
+		list INTERFACE_LIST {  // 1st list
+			key ifname;
+			leaf ifname {
+				type string {
+					pattern "Eth.*";
+				}
+			}
+			// ...
+		}
+
+		list INTERFACE_ANOTHER_LIST { // Negative case
+			key ifname;
+			leaf ifname {
+				type string {
+					pattern "Eth.*";
+				}
+			}
+			// ...
+		}
+	} 
+}
+......
+```
+
+#### Example 5: Length Constraints (Allowed case)
+
+In this example, `INTERFACE_LIST` uses a key ifname that must be exactly 3 characters long, while `INTERFACE_IPADDR_LIST` uses a key ifname that must be at least 4 characters long.
+
+```yang
+......
+container sonic-interface {
+	container INTERFACE {
+		list INTERFACE_LIST {  // 1st list
+			key ifname;
+			leaf ifname {
+				type string {
+					length "3..3";
+				}
+			}
+			// ...
+		}
+
+		list INTERFACE_IPADDR_LIST { // 2nd list
+			key "ifname ip_addr";
+			leaf ifname {
+				type string {
+					length "4..max";
+				}
+			}
+			// ...
+		}
+	} 
+}
+......
+```
+
+#### Example 6: Length Constraints (NOT Allowed case)
+
+Here, both `INTERFACE_LIST` and `INTERFACE_ANOTHER_LIST` use a key ifname with overlapping length constraints, making them potentially ambiguous.
```

Comment:
I have similar commands for examples, 4 to 6. Do we have valid use case right now, where these example or constraints can be used. Thanks.

Reply:
Looking at telemetry YANG, seems there is a genuine case. If yes then.

-- Update Guideline. [please provide real example of today's config].
-- above should be done after translation PR is raised, which supports pattern if primary key has 'same number of element and same type.

Reply:
@praveen-li IMO guideline PR has to go first then the translations. The translations are based on guideline. correct me.